### ISSUE TITLE
Allow to use SVG as icons

### ISSUE THREAD

**Issue by sn3w** [Alexays/Waybar#1330]
Icons like Octicons provide SVG files for setting up icons and I'll love to see Waybar support having SVG set as icons.

**Comment by Moelf** [comment]
would love to be able to use things like https://heroicons.com/

**Comment by usernamer-git** [comment]
Would love to use https://materialdesignicons.com/ in the waybar.

**Comment by hanckmann** [comment]
I would really like this as well.
What is required to get svg support in waybar. My issue is mainly the svg icons in the systray which do not show up.

**Comment by usernamer-git** [comment]
As workarround i found it is able to load svg as background image:
![waybar](https://user-images.githubusercontent.com/84246111/222962621-f635b8de-407a-4d85-9f55-e43c4af1148b.png)
```
// waybar config
...
"modules-left": [ "custom/logo", ... ]
...
"custom/logo": {
  "format": "    ",
  "tooltip": false,
  "on-click": "exec lain.sh &"
},
...
```

```
/* waybar style.css */
...
#custom-logo {
  background-image: url('/home/lain/.config/waybar/void.svg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain; 
}
...
```
i know this is not the best way to do, but I wanted to share this temporary method I'm using for now.

**Comment by sandmuel** [comment]
> As workarround i found it is able to load svg as background image: ![waybar](https://user-images.githubusercontent.com/84246111/222962621-f635b8de-407a-4d85-9f55-e43c4af1148b.png)
> 
> ```
> // waybar config
> ...
> "modules-left": [ "custom/logo", ... ]
> ...
> "custom/logo": {
>   "format": "    ",
>   "tooltip": false,
>   "on-click": "exec lain.sh &"
> },
> ...
> ```
> 
> ```
> /* waybar style.css */
> ...
> #custom-logo {
>   background-image: url('/home/lain/.config/waybar/void.svg');
>   background-position: center;
>   background-repeat: no-repeat;
>   background-size: contain; 
> }
> ...
> ```
> 
> i know this is not the best way to do, but I wanted to share this temporary method I'm using for now.

Any way to do something like this on a module with multiple states? (such as the battery module)

**Comment by v4u6h4n** [comment]
I would really like this feature, as doing something like using bitone/duotone icons is possible with fonts as far as I can tell, and it would make a massive difference in customisability.

**Comment by jlo62** [comment]
> Would love to use https://materialdesignicons.com/ in the waybar.
workaround: Arch has ttf-material-design-icons-git.

**Comment by sfuentesojeda** [comment]
I assume your workaround of using it as a background image doesn't allow to change it's color right?

**Comment by jlo62** [comment]
> allow to change it's color right?

Just change the color of the svg with, for example, a text editor or inkscape?

**Comment by sfuentesojeda** [comment]
> > allow to change it's color right?
> 
> Just change the color of the svg with, for example, a text editor or inkscape?
Sorry, I should've added that change the color of the svg dinamically, like with pywal or make the fill attribute usable to allow it being dinamic, but I suppose that's no easy task or limitations of gtk

**Comment by Lodobo** [comment]
@sfuentesojeda The correct way to do this would be to style the svg icon with CSS, just like you would in a browser.

**Comment by tom-gora** [comment]
Unfortunately theming anything gtk or pango related painfully reminds my that it is not a browser and, with all its faults and baggage,  how many light years ahead web is in terms of layout and styling compared to what is available in the native toolkits.

**Comment by DurableSteer** [comment]
I was thinking to contribute to fix this issue. 
I looked into it and the image module seems to work fine with svg files out of the box.
Am i misunderstanding the feature request or is this issue already solved but not marked as solved?

**Comment by Lodobo** [comment]
> I was thinking to contribute to fix this issue. I looked into it and the image module seems to work fine with svg files out of the box. Am i misunderstanding the feature request or is this issue already solved but not marked as solved?

Yes, the image module can load SVGs, but the other modules can not be styled with SVGs (as far as I am aware). Re-implementing the functionality of all the other modules with scripts through the image module is both tedious and impractical.

Maybe if there was a way to link the image module to another module.

**Comment by Muramasa500** [comment]
I wanted to add SVG as icons  to show active programs on each workspace, but unfortunately it seems not to support SVG. It can only show fonts from Nerd Fonts to show icons. Would be great if waybar could support SVG.
